Test failure output truncates stderr and mixes JSON messages, hiding root cause compilation errors

## Summary

When `cargo-wdk` build command tests (e.g. `kmdf_driver_cross_compiles_with_cli_option_successfully`) fail due to a compilation error, the test failure output has two issues that make debugging difficult:

1. **Stderr is truncated** to a summary like `<168 lines total>`, hiding the actual compiler error.
2. **JSON-formatted cargo messages are mixed into the output** (e.g. `{\"reason\":\"build-finished\",\"success\":false}`), making the output hard to read and obscuring human-readable error messages.

## Steps to Reproduce

1. Remove the `aarch64-pc-windows-msvc` Rust target (or run on a machine where it is not installed):
   ```
   rustup target remove aarch64-pc-windows-msvc
   ```
2. Run the test:
   ```
   cargo test -p cargo-wdk --test build_command_test -- kmdf_driver_cross_compiles_with_cli_option_successfully
   ```

## Actual Behavior

The test panics with output like:

```
thread 'kmdf_driver_cross_compiles_with_cli_option_successfully' panicked at ...:
  Unexpected failure.
  code=1
  stderr=`<168 lines total>
  `
   INFO Building package kmdf-driver
     Compiling proc-macro2 v1.0.101
     ...
      {\"reason\":\"build-finished\",\"success\":false}

ERROR One or more workspace members failed to build in the workspace: ...
```

Two problems:

- The actual compilation error (`error[E0463]: can't find crate for core ... the arch64-pc-windows-msvc target may not be installed`) is buried in the truncated 168 lines of stderr and never shown in the panic message.
- JSON-serialized cargo messages like `{\"reason\":\"build-finished\",\"success\":false}` appear interleaved with human-readable output, making the output confusing. These escaped JSON strings add noise and make it harder to spot the real error even if the output were not truncated.

## Expected Behavior

The test failure output should:

1. **Surface the actual compiler error(s)** so that the root cause is immediately visible. This could be done by:
   - Printing the full stderr on failure, or
   - Extracting and highlighting lines matching `error\[E\d+\]` or `^error` from stderr in the failure message, or
   - Increasing the `assert_cmd` output truncation limit for test assertions

2. **Present human-readable output** rather than raw JSON cargo messages. The JSON messages could be parsed and either filtered out or reformatted into readable text.

## Context

The test uses `assert_cmd`'s `.assert().success()` (in `run_build_cmd` at `build_command_test.rs:366`), which truncates long output by default. The JSON messages come from cargo's `--message-format=json` output being mixed into stderr. Together, these make test failures unnecessarily difficult to diagnose.

--- a/crates/cargo-wdk/src/actions/build/build_task.rs
+++ b/crates/cargo-wdk/src/actions/build/build_task.rs
@@ -17,6 +17,7 @@ use wdk_build::CpuArchitecture;
 use crate::providers::exec::CommandExec;
 use crate::{
     actions::{Profile, build::error::BuildTaskError, to_target_triple},
+    providers::error::CommandError,
     trace,
 };
 
@@ -121,8 +122,17 @@ impl<'a> BuildTask<'a> {
         let output = self
             .command_exec
             .run("cargo", &args, None, Some(self.working_dir))
-            .map_err(|_| {
-                BuildTaskError::CargoBuild("Compilation failed. See compiler errors above.".into())
+            .map_err(|err| match err {
+                CommandError::CommandFailed {
+                    command,
+                    args,
+                    stdout: _,
+                } => BuildTaskError::CargoBuild(CommandError::CommandFailed {
+                    command,
+                    args,
+                    stdout: String::new(), // omit stdout to avoid printing potentially large output
+                }),
+                err @ CommandError::IoError(..) => BuildTaskError::CargoBuild(err),
             })?;
 
         debug!("cargo build done");
@@ -269,7 +279,7 @@ mod tests {
     }
 
     #[test]
-    fn run_returns_error_when_cargo_command_fails() {
+    fn run_returns_command_failed_error_with_empty_stdout_when_cargo_build_exits_nonzero() {
         let working_dir = PathBuf::from("C:/abs/driver");
         let mut mock = MockCommandExec::new();
         mock.expect_run().return_once(|_, _, _, _| {
@@ -295,9 +305,58 @@ mod tests {
         );
 
         let err = task.run().err().expect("expected cargo failure");
-        let BuildTaskError::CargoBuild(msg) = err else {
-            panic!("expected cargo build error");
+        let BuildTaskError::CargoBuild(CommandError::CommandFailed {
+            command,
+            args,
+            stdout,
+        }) = err
+        else {
+            panic!("expected CargoBuild(CommandFailed) error, got: {err:?}");
         };
-        assert_eq!(msg, "Compilation failed. See compiler errors above.");
+        assert_eq!(command, "cargo");
+        assert!(
+            args.contains(&"build".to_string()),
+            "expected args to contain 'build', got: {args:?}"
+        );
+        assert!(
+            stdout.is_empty(),
+            "expected stdout to be omitted, got: {stdout}"
+        );
+    }
+
+    #[test]
+    fn run_returns_io_error_when_cargo_build_command_invocation_fails() {
+        let working_dir = PathBuf::from("C:/abs/driver");
+        let mut mock = MockCommandExec::new();
+        mock.expect_run().return_once(|_, _, _, _| {
+            Err(CommandError::from_io_error(
+                "cargo",
+                &["build"],
+                std::io::Error::new(std::io::ErrorKind::NotFound, "program not found"),
+            ))
+        });
+
+        let task = BuildTask::new(
+            "my-driver",
+            &working_dir,
+            None,
+            None,
+            clap_verbosity_flag::Verbosity::default(),
+            &mock,
+        );
+
+        let err = task
+            .run()
+            .err()
+            .expect("expected command invocation failure");
+        let BuildTaskError::CargoBuild(CommandError::IoError(command, args, io_err)) = err else {
+            panic!("expected CargoBuild(IoError) error, got: {err:?}");
+        };
+        assert_eq!(command, "cargo");
+        assert!(
+            args.contains(&"build".to_string()),
+            "expected args to contain 'build', got: {args:?}"
+        );
+        assert_eq!(io_err.kind(), std::io::ErrorKind::NotFound);
     }
 }

--- a/crates/cargo-wdk/src/actions/build/build_task.rs
+++ b/crates/cargo-wdk/src/actions/build/build_task.rs
@@ -122,17 +122,14 @@ impl<'a> BuildTask<'a> {
         let output = self
             .command_exec
             .run("cargo", &args, None, Some(self.working_dir))
-            .map_err(|err| match err {
-                CommandError::CommandFailed {
-                    command,
-                    args,
-                    stdout: _,
-                } => BuildTaskError::CargoBuild(CommandError::CommandFailed {
-                    command,
-                    args,
-                    stdout: String::new(), // omit stdout to avoid printing potentially large output
-                }),
-                err @ CommandError::IoError(..) => BuildTaskError::CargoBuild(err),
+            .map_err(|mut err| {
+                // Drop stdout from CommandFailed so the noisy
+                // --message-format=json-render-diagnostics output isn't bubbled up
+                // in the wrapped error.
+                if let CommandError::CommandFailed { stdout, .. } = &mut err {
+                    stdout.clear();
+                }
+                BuildTaskError::CargoBuild(err)
             })?;
 
         debug!("cargo build done");

--- a/crates/cargo-wdk/src/actions/build/build_task.rs
+++ b/crates/cargo-wdk/src/actions/build/build_task.rs
@@ -91,7 +91,7 @@ impl<'a> BuildTask<'a> {
     ) -> Result<impl Iterator<Item = Result<Message, std::io::Error>>, BuildTaskError> {
         debug!("Running cargo build");
         let mut args = vec!["build".to_string()];
-        args.push("--message-format=json".to_string());
+        args.push("--message-format=json-render-diagnostics".to_string());
         args.push("-p".to_string());
         args.push(self.package_name.to_string());
         if let Some(path) = self.manifest_path.to_str() {
@@ -120,7 +120,10 @@ impl<'a> BuildTask<'a> {
         // is respected
         let output = self
             .command_exec
-            .run("cargo", &args, None, Some(self.working_dir))?;
+            .run("cargo", &args, None, Some(self.working_dir))
+            .map_err(|_| {
+                BuildTaskError::CargoBuild("Compilation failed. See compiler errors above.".into())
+            })?;
 
         debug!("cargo build done");
         Ok(Message::parse_stream(std::io::Cursor::new(output.stdout)))
@@ -205,7 +208,7 @@ mod tests {
         let verbosity = clap_verbosity_flag::Verbosity::default();
         let expected_args = vec![
             "build".to_string(),
-            "--message-format=json".to_string(),
+            "--message-format=json-render-diagnostics".to_string(),
             "-p".to_string(),
             "my-driver".to_string(),
             "--manifest-path".to_string(),
@@ -292,22 +295,9 @@ mod tests {
         );
 
         let err = task.run().err().expect("expected cargo failure");
-        let BuildTaskError::CargoBuild(command_error) = err else {
+        let BuildTaskError::CargoBuild(msg) = err else {
             panic!("expected cargo build error");
         };
-        match command_error {
-            CommandError::CommandFailed {
-                command,
-                args,
-                stdout,
-            } => {
-                assert_eq!(command, "cargo");
-                assert_eq!(args, vec!["build".to_string()]);
-                assert_eq!(stdout, "error");
-            }
-            CommandError::IoError(_, _, err) => {
-                panic!("expected CommandFailed, got IoError: {err}")
-            }
-        }
+        assert_eq!(msg, "Compilation failed. See compiler errors above.");
     }
 }

--- a/crates/cargo-wdk/src/actions/build/error.rs
+++ b/crates/cargo-wdk/src/actions/build/error.rs
@@ -48,8 +48,8 @@ pub enum BuildActionError {
 pub enum BuildTaskError {
     #[error("Empty manifest path found error")]
     EmptyManifestPath,
-    #[error("Error running cargo build command: {0}")]
-    CargoBuild(String),
+    #[error("Error running cargo build command")]
+    CargoBuild(#[source] CommandError),
     #[error(transparent)]
     FileIo(#[from] FileError),
 }

--- a/crates/cargo-wdk/src/actions/build/error.rs
+++ b/crates/cargo-wdk/src/actions/build/error.rs
@@ -48,8 +48,8 @@ pub enum BuildActionError {
 pub enum BuildTaskError {
     #[error("Empty manifest path found error")]
     EmptyManifestPath,
-    #[error("Error running cargo build command")]
-    CargoBuild(#[from] CommandError),
+    #[error("Error running cargo build command: {0}")]
+    CargoBuild(String),
     #[error(transparent)]
     FileIo(#[from] FileError),
 }

--- a/crates/cargo-wdk/src/actions/build/tests.rs
+++ b/crates/cargo-wdk/src/actions/build/tests.rs
@@ -1935,7 +1935,7 @@ impl TestBuildAction {
             .to_string();
         let mut expected_cargo_build_args: Vec<String> = vec![
             "build",
-            "--message-format=json",
+            "--message-format=json-render-diagnostics",
             "-p",
             &driver_name,
             "--manifest-path",

